File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/normx/trim_edges.go
package normx

// TrimEdges normalizes with the trim edges rule.
func TrimEdges(text string) string {
	begin := 0
	end := len(text)
	for begin < end && (text[begin] == ' ' || text[begin] == '\t') {
		begin++
	}
	for end > begin && (text[end-1] == ' ' || text[end-1] == '\t') {
		end--
	}
	return text[begin:end]
}

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/TrimEdges.java
package com.freight.norm;

/** Normalizer: trim edges. */
public final class TrimEdges implements Normalizer {

    @Override
    public String name() {
        return "trim_edges";
    }

    @Override
    public String apply(String text) {
        int begin = 0;
        int end = text.length();
        while (begin < end && (text.charAt(begin) == ' ' || text.charAt(begin) == '\t')) {
            begin++;
        }
        while (end > begin && (text.charAt(end - 1) == ' ' || text.charAt(end - 1) == '\t')) {
            end--;
        }
        return text.substring(begin, end);
    }
}
